net/libfab/libfab.c

```diff
@@ -2593,39 +2597,46 @@ static int libfab_ping_op(struct m0_fab__active_ep *aep, struct m0_fab__buf *fb)
  * This function will call the bulk transfer operation (read/write) on the
  * net-buffer.
  */
-static int libfab_bulk_op(struct m0_fab__active_ep *aep, struct m0_fab__buf *fb)
+static int libfab_bulk_op(struct m0_fab__active_ep *aep, struct m0_fab__buf *fb,
+			  uint32_t max_iov)
 {
 	struct m0_fab__buf_xfer_params xp;
 	struct fi_msg_rma              op_msg;
 	struct fi_rma_iov             *r_iov;
-	struct fi_rma_iov              remote[FAB_MAX_IOV_PER_TX];
-	struct iovec                   loc_iv[FAB_MAX_IOV_PER_TX];
+	struct fi_rma_iov             *remote = NULL;
+	struct iovec                  *loc_iv = NULL;
 	m0_bcount_t                   *v_cnt;
 	uint64_t                       op_flag;
 	uint32_t                       loc_slen;
 	uint32_t                       rem_slen;
 	uint32_t                       wr_cnt = 0;
 	uint32_t                       idx;
-	uint32_t                       max_iov_per_tx;
 	int                            ret = 0;
 	bool                           isread;
 	bool                           last_seg = false;
 
-	M0_ENTRY("loc_buf=%p q=%d loc_seg=%d rem_buf=%d rem_seg=%d",
-		 fb, fb->fb_nb->nb_qtype, fb->fb_nb->nb_buffer.ov_vec.v_nr, 
-		 (int)fb->fb_rbd->fbd_buftoken, (int)fb->fb_rbd->fbd_iov_cnt);
+	M0_ENTRY("loc_buf=%p q=%d loc_seg=%d rem_buf=%d rem_seg=%d iov_max=%d",
+		 fb, fb->fb_nb->nb_qtype, fb->fb_nb->nb_buffer.ov_vec.v_nr,
+		 (int)fb->fb_rbd->fbd_buftoken, (int)fb->fb_rbd->fbd_iov_cnt,
+		 (int)max_iov);
 	M0_PRE(fb->fb_rbd != NULL);
 
+	M0_ALLOC_ARR(remote, max_iov);
```

Comment:
For optimization instead of allocating and freeing we can allocate at time of libfab_passive_ep_create and free it when ep  is getting deleted?

Reply:
Done.